validator/client/src/main/java/tech/pegasys/artemis/validator/client/Signer.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.validator.client;
+
+import static tech.pegasys.artemis.datastructures.util.BeaconStateUtil.compute_signing_root;
+import static tech.pegasys.artemis.datastructures.util.BeaconStateUtil.get_domain;
+
+import com.google.common.primitives.UnsignedLong;
+import org.apache.tuweni.bytes.Bytes;
+import tech.pegasys.artemis.datastructures.state.Fork;
+import tech.pegasys.artemis.datastructures.validator.MessageSignerService;
+import tech.pegasys.artemis.util.async.SafeFuture;
+import tech.pegasys.artemis.util.bls.BLSSignature;
+import tech.pegasys.artemis.util.config.Constants;
+
+public class Signer {
+  private final MessageSignerService signerService;
+
+  public Signer(final MessageSignerService signerService) {
+    this.signerService = signerService;
+  }
+
+  public SafeFuture<BLSSignature> createRandaoReveal(final UnsignedLong epoch, final Fork fork) {
+    Bytes domain = get_domain(Constants.DOMAIN_RANDAO, epoch, fork);
+    Bytes signing_root = compute_signing_root(epoch.longValue(), domain);
+    return signerService.sign(signing_root);
+  }
```

Comment:
Note no access to `BeaconState` in the validator client so can't use the existing `get_epoch_signature`.

Reply:
Is this necessary duplication?

Reply:
The original version will wind up being removed.  The beacon chain has no need to create randao reveal - the validator client provides it in the request to create a block.